File: BASICO/Reto_8/Solucion_1.cpp
#include <iostream>
#include <string>
using namespace std;
string romans(int n) {
    int values[] = {1000,900,500,400,100,90,50,40,10,9,5,4,1};
    string symbols[] = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
    string result = "";
    int size = sizeof(values)/sizeof(values[0]);
    for (int i = 0; i < size; i++) {
        while (n >= values[i]) {
            result += symbols[i];
            n -= values[i];
        }
    }
    return result;
}
int main() {
    int number;
    cout <<"Enter a number: "; cin >> number;
    cout <<"Result: "<< romans(number);
    return 0;
}

File: BASICO/Reto_8/Solucion_1.py
def romans(n):
    values = [
        (1000,"M"), (900,"CM"), (500,"D"), (400,"CD"),
        (100,"C"), (90,"XC"), (50,"L"), (40,"XL"),
        (10,"X"), (9,"IX"), (5,"V"), (4,"IV"), (1,"I")
    ]
    result= ""
    for worth, symbol in values:
        while n >= worth:
            result += symbol
            n -= worth
    return result
number = int(input("Enter a number: "))
print("Result: ", romans(number))